src/trans.rs

```diff
@@ -1310,25 +1337,12 @@ impl<'v, 'tcx: 'v, 'module: 'v> BinaryenFnCtxt<'v, 'tcx, 'module> {
     }
 
     fn trans_lval(&mut self, lvalue: &Lvalue<'tcx>) -> Option<BinaryenLvalue> {
+        let mir = self.mir.borrow();
+
         let i = match *lvalue {
-            Lvalue::Arg(i) => i.index() as u32,
-            Lvalue::Var(i) => {
+            Lvalue::Local(i) => {
```

Comment:
Okay, awesome! This seems way cleaner.

We'll need to combine `self.var_map` and `self.temp_map` then, in lines 257 and 267, to match the new flat indexing.

Basically, these maps keep track of the Wasm type corresponding to each local variables, assuming they have a type. As it is now, I expect the `var_map` lookup in the new line 1344 will fail for temps, since we only added variables to it.